src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/ccd/CcdClient.java

```diff
@@ -43,30 +44,35 @@ public void completeAwaitingDcnProcessing(
         String caseTypeId = service.toUpperCase() + "_ExceptionRecord";
         String eventId = COMPLETE_AWAITING_DCN_PROCESSING_EVENT_ID;
 
-        StartEventResponse startEventResponse =
-            startEvent(authenticator, jurisdiction, caseTypeId, exceptionRecordCcdId, eventId);
-
-        Event event = Event
-            .builder()
-            .summary("Complete payment DCN processing")
-            .id(startEventResponse.getEventId())
-            .build();
-
-        CaseDataContent caseDataContent = CaseDataContent.builder()
-            .data(ImmutableMap.of(AWAITING_DCN_PROCESSING_FIELD_NAME, "No"))
-            .event(event)
-            .ignoreWarning(true)
-            .caseReference(exceptionRecordCcdId)
-            .eventToken(startEventResponse.getToken())
-            .build();
-
-        submitEvent(authenticator, jurisdiction, caseTypeId, exceptionRecordCcdId, caseDataContent);
-
-        log.info(
-            "Completed awaiting payment DCN processing. Exception record ID: {}, service {}",
-            exceptionRecordCcdId,
-            service
-        );
+        try {
+            StartEventResponse startEventResponse =
+                startEvent(authenticator, jurisdiction, caseTypeId, exceptionRecordCcdId, eventId);
+
+            Event event = Event
+                .builder()
+                .summary("Complete payment DCN processing")
+                .id(startEventResponse.getEventId())
+                .build();
+
+            CaseDataContent caseDataContent = CaseDataContent.builder()
+                .data(ImmutableMap.of(AWAITING_DCN_PROCESSING_FIELD_NAME, "No"))
+                .event(event)
+                .ignoreWarning(true)
+                .caseReference(exceptionRecordCcdId)
+                .eventToken(startEventResponse.getToken())
+                .build();
+
+            submitEvent(authenticator, jurisdiction, caseTypeId, exceptionRecordCcdId, caseDataContent);
+
+            log.info(
+                "Completed awaiting payment DCN processing. Exception record ID: {}, service {}",
+                exceptionRecordCcdId,
+                service
+            );
+        } catch (FeignException ex) {
+            log.error("Feign exception thrown. Body: {}", ex.contentUTF8(), ex);
+            throw ex;
+        }
```

Comment:
This whole change is there for logging error responses while troubleshooting. This should be removed from this PR and in a future PR we can add the logging conditionally (can't log this in prod) and with a more descriptive message.

Reply:
I suggest to leave this here. This will make life easier in case of potential future failures. Otherwise we will have to change code each time the test fails moreover it is an external call. Also this is a test code - does not need to be minimalistic.

Reply:
If we leave this here, then we need to change two things:
- make it conditional - can't run in prod
- make it more descriptive. Feign exception thrown` - that was fine for troubleshooting in tests

This is application code.

Reply:
Let's revert this change, this PR is about adding functional tests after all, this is not needed for those tests if I understand correctly.
Can be added on a separate PR and discussed there.

Reply:
Logging as a separate change would be ideal. After some discussion, logging the response body doesn't even have to be turned off in production, as we're not sending any sensitive data here. We will need to treat `contentUTF8` safely, because if there's no content, that method will throw an exception. We should replace `contentUTF8` with something like this: `e.content() != null ? e.contentUTF8() : null`